app/src/main/java/com/cornellappdev/transit/ui/components/RouteCell.kt

```diff
@@ -0,0 +1,682 @@
+                                bottom.linkTo(startLoc.bottom)
+                                start.linkTo(busBox.end, margin = 16.dp)
+                            })
+
+                        Icon(
+                            imageVector = ImageVector.vectorResource(R.drawable.bus_route_line),
+                            tint = TransitBlue,
+                            contentDescription = "",
+                            modifier = Modifier.constrainAs(line) {
+                                top.linkTo(startIcon.top)
+                                bottom.linkTo(endIcon.bottom)
+                                end.linkTo(startIcon.end)
+                                start.linkTo(startIcon.start)
+
+                            })
+
+                        Icon(
+                            imageVector = ImageVector.vectorResource(R.drawable.boarding_stop),
+                            tint = Color.Unspecified,
+                            contentDescription = "",
+                            modifier = Modifier.constrainAs(endIcon) {
+                                end.linkTo(startIcon.end)
+                                top.linkTo(endStop.top)
+                                bottom.linkTo(endStop.bottom)
+                                start.linkTo(startIcon.start)
+                            })
+
+                        Icon(
+                            imageVector = ImageVector.vectorResource(R.drawable.bus_route_line),
+                            tint = TransitBlue,
+                            contentDescription = "",
+                            modifier = Modifier.constrainAs(line2) {
+                                top.linkTo(endIcon.top)
+                                bottom.linkTo(destIcon.bottom)
+                                end.linkTo(endIcon.end)
+                                start.linkTo(endIcon.start)
+
+                            })
+
+                        Icon(
+                            imageVector = ImageVector.vectorResource(R.drawable.bus_destination),
+                            tint = Color.Unspecified,
+                            contentDescription = "",
+                            modifier = Modifier.constrainAs(destIcon) {
+                                top.linkTo(destLoc.top)
+                                bottom.linkTo(destLoc.bottom)
+                                end.linkTo(startIcon.end)
+                                start.linkTo(startIcon.start)
+                            })
+
+
+                    }
+
+                }
+            }
+        }
+    }
+
+}
+
+enum class BusLateness {
+    LATE {
+        override fun color() = LateRed
+        override fun text() = "Delayed"
+    },
+    NORMAL {
+        override fun color() = LiveGreen
+        override fun text() = "On Time"
+    },
+    NONE {
+        override fun color() = Color.Black
+        override fun text() = ""
+    };
+
+    abstract fun color(): Color
+    abstract fun text(): String
+}
```

Comment:
Ooh that's some funky syntax but I like it! Seems to work well. 

Also throw this in its own file, though. UI Component files should only have UI component stuff.